### PR TITLE
[bs_parser_hevc] Add required checks

### PR DESCRIPTION
**Pull request by ndomnina**

### CODE REVIEW

File: tools/bs_parser_hevc/src/hevc2_parser.cpp [hunk 1]
```diff
@@ -759,7 +759,10 @@ Bs32u Parser::ParseSSDSubmit(SDThread*& pSDT, NALU* AU)
 
         sdpar.Emulation = true;
 
-        TargetRBSP = Bs32u(CurRBSP / m_sdt.size());
+        if (m_sdt.empty())
```
Comment (mgonchar) [inline tools/bs_parser_hevc/src/hevc2_parser.cpp]: no need of this, I synced with Evgeniy, size of vector can't be zero here

File: tools/bs_parser_hevc/src/hevc2_parser.cpp [hunk 2]
```diff
@@ -875,7 +878,11 @@ Bs32u Parser::ParseSSDSubmit(SDThread*& pSDT, NALU* AU)
         if (sdpar.ColPic)
         {
             lock(sdpar.ColPic);
-            sliceDep.push_back(spColPic);
+
+            if (spColPic)
```
Comment (mgonchar) [inline tools/bs_parser_hevc/src/hevc2_parser.cpp]: @eistarov do we need this check?

File: tools/bs_parser_hevc/src/hevc2_parser.cpp (hunk 1 above)
Comment (ndomnina) [reply tools/bs_parser_hevc/src/hevc2_parser.cpp]: This check was removed

File: tools/bs_parser_hevc/src/hevc2_parser.cpp (hunk 2 above)
Comment (ndomnina) [reply tools/bs_parser_hevc/src/hevc2_parser.cpp]: This check was removed

File: tools/bs_parser_hevc/tools/hevc_fei_extractor/src/main.cpp [hunk 3]
```diff
@@ -67,7 +67,7 @@ class FileHandler
             m_errorSts = true;// Error sts
         }
 
-        if ((m_pFile = fopen(fileName, mode)) == nullptr) {
+        if (fileName && (m_pFile = fopen(fileName, mode)) == nullptr) {
```
Comment (mgonchar) [inline tools/bs_parser_hevc/tools/hevc_fei_extractor/src/main.cpp]: filename is checked at line 60 above. No need to check again

Comment (ndomnina) [reply tools/bs_parser_hevc/tools/hevc_fei_extractor/src/main.cpp]: But we do not exit the function after check. If the file name is empty, we still try to open the file.

Comment (ndomnina) [reply tools/bs_parser_hevc/tools/hevc_fei_extractor/src/main.cpp]: Fixed

**Pull request merged**